### ISSUE TITLE
Uncaught TypeError: path must be a string or Buffer

### ISSUE THREAD

**Issue by arnokret** [James-Yu/Atom-LaTeX#111]
When trying to compile my latex document on windows 10, I get the above error message. The issue seems also to be discussed in https://github.com/James-Yu/Atom-LaTeX/pull/47 , although there is no solution presented there? 

Please note I am not too familiar with the technical details of your project. I tried a couple of things, like setting a %!TEX root= ...  but it seems to have no effect. 

What am I doing wrong?

**Comment by ashthespy** [comment]
Could you kindly provide the entire issue log?

**Comment by vmetsis** [comment]
I got the same error, on a different situation.
===
[Enter steps to reproduce:]

1. Ctrl + Left mouse click on PDF viewer to sync with the corresponding position on a latex file. 
Notes: 
The exception occured on an Ubuntu 16.04 machine. The PDF was originally built on an other Mac OS machine.
The problem went away after cleaning all build files and rebuilding the PDF.


**Atom**: 1.19.4 x64
**Electron**: 1.6.9
**OS**: Ubuntu 16.04.3
**Thrown From**: [atom-latex](https://github.com/James-Yu/Atom-LaTeX) package 0.8.1


### Stack Trace

Uncaught TypeError: path must be a string or Buffer

```
At fs.js:871

TypeError: path must be a string or Buffer
    at Object.fs.readdirSync (fs.js:871:18)
    at Object.fs.readdirSync (ELECTRON_ASAR.js:576:28)
    at Manager.module.exports.Manager.loadLocalCfg (/packages/atom-latex/lib/manager.coffee:33:5)
    at Manager.module.exports.Manager.findMainConfig (/packages/atom-latex/lib/manager.coffee:123:6)
    at Manager.module.exports.Manager.findMainSequence (/packages/atom-latex/lib/manager.coffee:86:21)
    at Manager.module.exports.Manager.findMain (/packages/atom-latex/lib/manager.coffee:53:15)
    at Manager.module.exports.Manager.findPDF (/packages/atom-latex/lib/manager.coffee:144:10)
    at Locator.module.exports.Locator.locate (/packages/atom-latex/lib/locator.coffee:49:31)
    at Viewer.module.exports.Viewer.wsHandler (/packages/atom-latex/lib/viewer.coffee:29:24)
    at /packages/atom-latex/lib/server.coffee:23:47)
    at emitTwo (events.js:106:13)
    at WebSocket.emit (events.js:191:7)
    at Receiver.ontext (/packages/atom-latex/node_modules/ws/lib/WebSocket.js:841:10)
    at /packages/atom-latex/node_modules/ws/lib/Receiver.js:536:18
    at /packages/atom-latex/node_modules/ws/lib/Receiver.js:368:7
    at /packages/atom-latex/node_modules/ws/lib/PerMessageDeflate.js:249:5
    at afterWrite (_stream_writable.js:386:3)
    at onwrite (_stream_writable.js:377:7)
    at afterTransform (_stream_transform.js:79:3)
    at TransformState.afterTransform (_stream_transform.js:54:12)
    at Zlib.callback (zlib.js:625:5)
```

### Commands

```
 27x -3:27.8.0 core:backspace (input.hidden-input)
     -3:12.1.0 atom-latex:preview (input.hidden-input)
 18x -2:03.4.0 core:select-right (input.hidden-input)
  2x -0:49.1.0 intentions:highlight (input.hidden-input)
```

### Non-Core Packages

```
atom-latex 0.8.1 
intentions 1.1.5 
```

**Comment by ashthespy** [comment]
@vmetsis how many projects did you have open in Atom at the point of this error, and what kind of project structure do you utilise?

**Comment by vmetsis** [comment]
I had just one project open as far as I can remember. The project is broken down into multiple Latex files. A main.tex file and a few other .tex files which are imported using the \input{...} command.

**Comment by LFelipeGomez** [comment]
Here I got same error. I just installed the package and run a test file [Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.20.0 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.6
**Thrown From**: [atom-latex](https://github.com/James-Yu/Atom-LaTeX) package 0.8.1


### Stack Trace

Uncaught TypeError: path must be a string or Buffer

```
At fs.js:871

TypeError: path must be a string or Buffer
    at Object.fs.readdirSync (fs.js:871:18)
    at Object.fs.readdirSync (ELECTRON_ASAR.js:576:28)
    at Manager.module.exports.Manager.loadLocalCfg (/packages/atom-latex/lib/manager.coffee:33:5)
    at Builder.module.exports.Builder.setCmds (/packages/atom-latex/lib/builder.coffee:142:20)
    at Builder.module.exports.Builder.build (/packages/atom-latex/lib/builder.coffee:16:6)
    at HTMLElement.click (/packages/atom-latex/lib/view/panel.js:106:85)
```

### Commands

```
     -3:54.2.0 atom-latex:synctex (a.stack-toggle)
```

### Non-Core Packages

```
atom-latex 0.8.1 
git-control 0.9.0 
Hydrogen 1.22.0 
```

**Comment by HanKruiger** [comment]
I had the same, issue. I figured it was caused by the fact that the LaTeX file wasn't in any of my projects, so to fix it, I did:

* Open the .tex file in a separate Atom session, so that it is within a project.
* Build LaTeX in Atom.

This does build the document successfully. I don't think this behaviour is intended, so it's probably a bug.

**Comment by ashthespy** [comment]
@HanKruiger  I do not follow - How did you set the LaTeX [root file ](https://github.com/James-Yu/Atom-LaTeX#setting-the-latex-root-file) when you ran into this error?

**Comment by vmetsis** [comment]
Same error occurred when clicking on the option "Clean Latex auxiliary files" on Atom-Latex.

[Enter steps to reproduce:]

1. Clicked on "Clean Latex auxiliary files"
2. ...

**Atom**: 1.19.4 x64
**Electron**: 1.6.9
**OS**: Ubuntu 16.04.3
**Thrown From**: [atom-latex](https://github.com/James-Yu/Atom-LaTeX) package 0.8.1


### Stack Trace

Uncaught TypeError: path must be a string or Buffer

```
At fs.js:871

TypeError: path must be a string or Buffer
    at Object.fs.readdirSync (fs.js:871:18)
    at Object.fs.readdirSync (ELECTRON_ASAR.js:576:28)
    at Manager.module.exports.Manager.loadLocalCfg (/packages/atom-latex/lib/manager.coffee:33:5)
    at Manager.module.exports.Manager.findMainConfig (/packages/atom-latex/lib/manager.coffee:123:6)
    at Manager.module.exports.Manager.findMainSequence (/packages/atom-latex/lib/manager.coffee:86:21)
    at Manager.module.exports.Manager.findMain (/packages/atom-latex/lib/manager.coffee:53:15)
    at Cleaner.module.exports.Cleaner.clean (/packages/atom-latex/lib/cleaner.coffee:12:24)
    at /packages/atom-latex/lib/main.coffee:35:48)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app/src/command-registry.js:265:35)
    at CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app/src/command-registry.js:3:65)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/usr/share/atom/resources/app/src/command-registry.js:166:25)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchApplicationMenuCommand (/usr/share/atom/resources/app/src/atom-environment.js:1338:34)
    at EventEmitter.outerCallback (/usr/share/atom/resources/app/src/application-delegate.js:334:31)
    at emitThree (events.js:116:13)
    at EventEmitter.emit (events.js:194:7)
```

### Commands

```
     -0:51.6.0 tabs:close-other-tabs (div.title)
     -0:36.8.0 atom-latex:clean (input.hidden-input)
```

### Non-Core Packages

```
atom-latex 0.8.1 
intentions 1.1.5 
```

**Comment by ashthespy** [comment]
@vmetsis Can you give me information about your project structure?

**Comment by HanKruiger** [comment]
@ashthespy When I *did* ran into the error, I tried setting the LaTeX root with the magic comment:
```
% !TEX root = \path\to\root\file.tex
```

However, since I just used one .tex file, it shouldn't have been necessary for me to set the root file, right?

**Comment by vmetsis** [comment]
Regarding my previous post, the project structure is a single latex file. The file was previously compiled on a different machine and the auxiliary files were copied to the current machine. It seems to me that the error is caused by an old file path inside one of the auxiliary files.

**Comment by ashthespy** [comment]
I did some digging around and was able to reproduce this error! 😁 

The package normally expects the LaTeX file to be in it's own Atom Project i,e 
```
└── Document_1
    ├── Img
    ├── .latexcfg
    ├── Document_1.tex
    └── Document_1_Chapter1.tex 
```
Then, multiple attempts are used to try and determine the main LaTeX file.

However, to reproduce this error I did:
1. Open a new Atom window, and then a new editor
1. Paste some LaTeX file,
1. Build from Active Editor
1. Get this error

Will fix this, sorry everyone!
